python/tests/sql_aggregates/test_average.py

```diff
@@ -0,0 +1,183 @@
+        self.execute_query(pipeline_name, expected_data, table_name, view_query)  
+
+class Avg_Where_Groupby(TestAggregatesBase):
+    def test_avg_where_groupy0(self):
+        pipeline_name ="test_avg_where_groupby0"
+        # validated using postgres
+        new_data = [
+            {"id": 0,"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"id": 1,"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"id" :1,"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        self.add_data(new_data)
+        expected_data = [{"id": 1, "c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        table_name = "avg_where_groupby0"
+        view_query = f'''CREATE VIEW avg_view AS SELECT
+                            id, AVG(c1) AS c1, AVG(c2) AS c2, AVG(c3) AS c3, AVG(c4) AS c4, AVG(c5) AS c5, AVG(c6) AS c6, AVG(c7) AS c7, AVG(c8) AS c8
+                        FROM {table_name}
+                        WHERE 
+                            c1 is NOT NULl AND c3 is NOT NULL AND c5 is NOT NULL AND c7 is NOT NULL
+                        GROUP BY id;'''   
+        self.execute_query(pipeline_name, expected_data, table_name, view_query) 
+        
+    def test_avg_where_groupby1(self):
+        pipeline_name ="test_avg_where_groupby1"
+        # checked manually
+        new_data = [
+            {"id": 0,"c1": 1, "c2": 2, "c3": 3, "c4": 4, "c5": 5, "c6": 6, "c7": None, "c8": 8},
+            {"id": 0,"c1": 11, "c2": 22, "c3": None, "c4": 45, "c5": 51, "c6": 6, "c7": 72, "c8": 88},
+            {"id": 1,"c1": None, "c2": 2, "c3": 30, "c4": 4, "c5": None, "c6": 60, "c7": 70, "c8": 8},
+            {"id" :1,"c1": 14, "c2": 21, "c3": 44, "c4": 32, "c5": 34, "c6": 90, "c7": 29, "c8": 67}]
+        self.add_data(new_data)
+        expected_data = [
+            {"id": 0,"avg_c1": 11, "avg_c2": 22, "avg_c3": None, "avg_c4": 45, "avg_c5": 51, "avg_c6": 6, "avg_c7": 72, "avg_c8": 88},
+            {"id" :1,"avg_c1": 14, "avg_c2": 21, "avg_c3": 44, "avg_c4": 32, "avg_c5": 34, "avg_c6": 90, "avg_c7": 29, "avg_c8": 67}]
+        table_name = "avg_where_groupby1"
+        view_query = f'''CREATE VIEW avg_view AS
+                         WITH avg_val AS (
+                            SELECT
+                                AVG(c1) AS avg_c1, AVG(c2) AS avg_c2, AVG(c3) AS avg_c3, AVG(c4) AS avg_c4, AVG(c5) AS avg_c5, AVG(c6) AS avg_c6, AVG(c7) AS avg_c7, AVG(c8) AS avg_c8
+                            FROM {table_name})
+                            SELECT
+                                t.id, AVG(t.c1) AS avg_c1, AVG(t.c2) AS avg_c2, AVG(t.c3) AS avg_c3, AVG(t.c4) AS avg_c4,  AVG(t.c5) AS avg_c5, AVG(t.c6) AS avg_c6, AVG(t.c7) AS avg_c7, AVG(t.c8) AS avg_c8
+                            FROM {table_name} t
+                            WHERE t.c4 > (SELECT avg_c4 FROM avg_val)
```

Comment:
you can keep this test, since you wrote it, but in this issue we are not looking to test correlated subqueries as here